native/src/main/java/io/ballerina/stdlib/nats/jetstream/listener/StreamMessageHandler.java

```diff
@@ -0,0 +1,193 @@
+            } else {
+                args2[0] = populatedMsgRecord;
+            }
+            args2[1] = true;
+            args2[2] = callerObj;
+            args2[3] = true;
+            dispatch(args2, msg.getSubject(), returnType);
+        } else {
+            throw Utils.createNatsError("Invalid remote function signature");
+        }
+    }
+
+    private BMap<BString, Object> getReadonlyMessage(Message msg) {
+        Map<String, Object> valueMap = new HashMap<>();
+        valueMap.put(Constants.MESSAGE_CONTENT, ValueCreator.createArrayValue(msg.getData()));
+        valueMap.put(Constants.MESSAGE_SUBJECT, StringUtils.fromString(msg.getSubject()));
+        return ValueCreator.createReadonlyRecordValue(Utils.getModule(), Constants.STREAM_MESSAGE, valueMap);
+    }
+
+    private void dispatch(Object[] args, String subject, Type returnType) {
+        executeResource(subject, args, returnType);
+    }
+
+    private void executeResource(String subject, Object[] args, Type returnType) {
+        CountDownLatch countDownLatch = new CountDownLatch(1);
+        ObjectType objectType = (ObjectType) TypeUtils.getReferredType(service.getType());
+        StrandMetadata metadata = new StrandMetadata(Utils.getModule().getOrg(), Utils.getModule().getName(),
+                Utils.getModule().getVersion(), Constants.ON_MESSAGE_RESOURCE);
+        if (ObserveUtils.isTracingEnabled()) {
+            Map<String, Object> properties = new HashMap<>();
+            NatsObserverContext observerContext = new NatsObserverContext(NatsObservabilityConstants.CONTEXT_CONSUMER,
+                    connectedUrl, subject);
+            properties.put(ObservabilityConstants.KEY_OBSERVER_CONTEXT, observerContext);
+            if (objectType.isIsolated() &&
+                    objectType.isIsolated(Constants.ON_MESSAGE_RESOURCE)) {
+                runtime.invokeMethodAsyncConcurrently(service, Constants.ON_MESSAGE_RESOURCE, null, metadata,
+                        new DispatcherCallback(connectedUrl, subject, countDownLatch), properties, returnType, args);
+            } else {
+                runtime.invokeMethodAsyncSequentially(service, Constants.ON_MESSAGE_RESOURCE, null, metadata,
+                        new DispatcherCallback(connectedUrl, subject, countDownLatch), properties, returnType, args);
+            }
+        } else {
+            if (objectType.isIsolated() &&
+                    objectType.isIsolated(Constants.ON_MESSAGE_RESOURCE)) {
+                runtime.invokeMethodAsyncConcurrently(service, Constants.ON_MESSAGE_RESOURCE, null, metadata,
+                        new DispatcherCallback(connectedUrl, subject, countDownLatch), null, returnType, args);
+            } else {
+                runtime.invokeMethodAsyncSequentially(service, Constants.ON_MESSAGE_RESOURCE, null, metadata,
+                        new DispatcherCallback(connectedUrl, subject, countDownLatch), null, returnType, args);
+            }
+        }
```

Comment:
Can't we simplify this like this?

```suggestion
        Map<String, Object> properties = null;
        if (ObserveUtils.isTracingEnabled()) {
            properties = new HashMap<>();
            NatsObserverContext observerContext = new NatsObserverContext(NatsObservabilityConstants.CONTEXT_CONSUMER,
                    connectedUrl, subject);
            properties.put(ObservabilityConstants.KEY_OBSERVER_CONTEXT, observerContext);
        }
        if (objectType.isIsolated() && objectType.isIsolated(Constants.ON_MESSAGE_RESOURCE)) {
            runtime.invokeMethodAsyncConcurrently(service, Constants.ON_MESSAGE_RESOURCE, null, metadata,
                    new DispatcherCallback(connectedUrl, subject, countDownLatch), null, returnType, args);
        } else {
            runtime.invokeMethodAsyncSequentially(service, Constants.ON_MESSAGE_RESOURCE, null, metadata,
                    new DispatcherCallback(connectedUrl, subject, countDownLatch), null, returnType, args);
        }
```